### PR TITLE
provider/azure: use simplestreams for images

### PR DESCRIPTION
**Pull request by axw**
Update the Azure provider to use simplestreams
for identifying images. This requires changes
to both the image and agent simplestreams, as
we will now use the current endpoint values.

NOTE: this _must_ not land until the simplestreams
changes have been made. If this lands first, then
the provider will be unusable.

### CODE REVIEW

**Comment by axw** [comment]
**QA**

1. bootstrap with custom simplestreams using --metadata-source, with the current region/endpoint values, and IDs in the Azure image URN format (publisher:offer:sku:version).
2. upload win2012 agent using github.com/axw/juju-tools
3. add-machine --series=win2012

File: provider/azure/environ.go [hunk 1]
```diff
@@ -1412,12 +1412,10 @@ func (env *azureEnviron) getStorageAccountKeyLocked(accountName string, refresh
 // TODO(axw) 2016-04-11 #1568715
 // When we have image simplestreams, we should rename this to "Region",
 // to implement simplestreams.HasRegion.
```
Comment (mjs) [inline provider/azure/environ.go]: Remove this TODO now?

File: provider/azure/internal/imageutils/images.go [hunk 2]
```diff
@@ -1,218 +0,0 @@
-// Copyright 2015 Canonical Ltd.
```
Comment (mjs) [inline provider/azure/internal/imageutils/images.go]: Niiiice :)

**Comment by mjs** [comment]
This is still waiting for simplestreams updates

**Pull request closed without merging**